torchtitan/utils.py

```diff
@@ -101,6 +102,55 @@ def run(self, step_count):
 SKIP_CLEANUP = "3"
 
 
+def create_context_parallel_ctx(
+    cp_mesh: DeviceMesh,
+    cp_buffers: List[torch.Tensor],
+    cp_seq_dims: List[int],
+    cp_no_restore_buffers: Set[torch.Tensor],
+):
+    try:
+        from torch.distributed.tensor.experimental import context_parallel
+    except ImportError:
+        print(
+            f"PyTorch version {torch.__version__} does not include the experimental "
+            "Context Parallel API. Please update to a newer version."
+        )
+
+    return context_parallel(
+        cp_mesh,
+        buffers=cp_buffers,
+        buffer_seq_dims=cp_seq_dims,
+        no_restore_buffers=cp_no_restore_buffers,
+    )
+
```

Comment:
nit: two empty lines in between, not sure if it's covered by `pre-commit`.
btw it seems linting test failed.